vision/use-cases/hey_llm/src/main.ts

```diff
@@ -0,0 +1,506 @@
+  const metadata = {
+    name: filename,
+    mimeType: mimeType,
+    parents: [parentFolderID],
+  };
+  const payload = `--foo_bar_baz
+Content-Type: application/json; charset=UTF-8
+
+${JSON.stringify(metadata)}
+
+--foo_bar_baz
+Content-Type: ${mimeType}
+Content-Transfer-Encoding: base64
+
+${base64image}
+
+--foo_bar_baz--`;
+  const uploadRes = UrlFetchApp.fetch(
+    'https://www.googleapis.com/upload/drive/v3/files?uploadType=multipart&fields=thumbnailLink',
+    {
+      method: 'post',
+      headers: {
+        Authorization: 'Bearer ' + oauth.getAccessToken(),
+        'Content-Type': 'multipart/related; boundary=foo_bar_baz',
+      },
+      payload: payload,
+      muteHttpExceptions: true,
+    },
+  );
+  const uploadResult: {thumbnailLink: string} = JSON.parse(
+    uploadRes.getContentText(),
+  );
+  return uploadResult.thumbnailLink;
+}
+
+/**
+ * Gets the GAS project name which this script belongs to.
+ * @private
+ * @param oauth OAuth2 Service
+ * @returns The script name
+ */
+function getScriptName_(oauth: GoogleAppsScriptOAuth2.OAuth2Service) {
+  const scriptId = ScriptApp.getScriptId();
+  const url = `https://www.googleapis.com/drive/v3/files/${scriptId}?fields=name`;
+  const response = UrlFetchApp.fetch(url, {
+    headers: {
+      Authorization: 'Bearer ' + oauth.getAccessToken(),
+    },
+  });
+  const data: {name: string} = JSON.parse(response.getContentText());
+  return data.name;
+}
+
+/**
+ * Creates a Drive folder for the GAS project.
+ * @private
+ * @param oauth OAuth2 Service
+ * @returns ID for the created Drive folder
+ */
+function createDriveFolder_(oauth: GoogleAppsScriptOAuth2.OAuth2Service) {
+  const metadata = {
+    name: `Generated images for: ${getScriptName_(oauth)}`,
+    mimeType: 'application/vnd.google-apps.folder',
+    parents: ['root'],
+  };
+  const payload = JSON.stringify(metadata);
+  const res = UrlFetchApp.fetch('https://www.googleapis.com/drive/v3/files', {
+    method: 'post',
+    headers: {
+      Authorization: 'Bearer ' + oauth.getAccessToken(),
+      'Content-Type': 'application/json',
+    },
+    payload: payload,
+    muteHttpExceptions: true,
+  });
+  const response: {id: string} = JSON.parse(res.getContentText());
+  return response.id;
+}
+
+/**
+ * Requests Vertex AI's Imagen to generate an image.
+ * @private
+ * @param oauth OAuth2 Service
+ * @param prompt A prompt for image generation
+ * @param seed A seed number
+ * @returns Generated result
+ */
+function requestImagen_(
+  oauth: GoogleAppsScriptOAuth2.OAuth2Service,
+  prompt: string,
+  seed: number,
+) {
+  const url = `https://${LOCATION}-aiplatform.googleapis.com/v1/projects/${getProjectNumber_()}/locations/${LOCATION}/publishers/google/models/${IMAGEN_MODEL}:predict`;
+
+  const payload = JSON.stringify({
+    instances: [
+      {
+        prompt: prompt,
+      },
+    ],
+    parameters: {
+      seed,
+      sampleCount: 1,
+      addWatermark: false,
+      safetySetting: 'block_few',
```

Comment:
Similar as above, can you use the Vertex AI JS SDK?

Reply:
With the same reason as the above comment, I need to construct the raw request to comply with AppScript's restriction in making external network calls via `UrlFetchApp`.